virttest/migration.py

```diff
@@ -271,26 +283,187 @@ def do_migration(self, vms, srcuri, desturi, migration_type,
         :param thread_timeout: time out seconds for the migration thread running
         :param ignore_status: determine if an exception is raised for errors
         :param func: the function executed during migration thread is running
+        :param multi_funcs: list of functions executed during migration
+                     thread is running. The func and multi_funcs should not be
+                     provided at same time.
         :param args: dictionary used by func,
                      'func_param' is mandatory if no real func_param, none is
                      requested.
                      'shell' is optional, where shell=True(bool) can be used
                      for process.run
 
         """
+
+        def _run_collect_event_cmd():
+            """
+            To execute virsh event command to collect the domain events
+
+            :return: VirshSession to retrieve the events
+            """
+            cmd = "event --loop --all"
+            virsh_event_session = virsh.VirshSession(virsh_exec=virsh.VIRSH_EXEC,
+                                                     auto_close=True,
+                                                     uri=srcuri)
+            virsh_event_session.sendline(cmd)
+            logging.debug("Begin to collect domain events...")
+            return virsh_event_session
+
+        def _need_collect_events(funcs_to_run):
+            """
+            Check if there is a need to run a command to collect domain events
+
+            :param funcs_to_run: the functions to be run. It can be a list
```

Comment:
Could you please add some comments to explain that the events will be collected if one of items has "after_event" or "before_event"?

Reply:
updated